samples/batch/transcription-enabled-storage/Connector/StorageUtilities.cs

```diff
@@ -0,0 +1,187 @@
+            {
+                receipt += $"\nEstimated cost: {Math.Round(audioDetails.EstimatedCost, 4)} €";
+                receipt += $"\nAudio length: {audioDetails.AudioLength}";
+            }
+            else
+            {
+                receipt += "\nUnable to parse file header.";
+            }
+
+            receipt += $"\nLocale: {audioDetails.Locale}";
+
+            // file storage
+            var storageAccount = CloudStorageAccount.Parse(storageConnectionString);
+            var fileClient = storageAccount.CreateCloudBlobClient();
+
+            // Get a reference to the file share we created previously.
+            var container = fileClient.GetContainerReference(containerName);
+            log.LogInformation("Got container reference for container: " + containerName);
+
+            var fileNameWithoutExtension = GetFileNameWithoutExtension(audioDetails.FileName);
+            var blockBlob = container.GetBlockBlobReference(fileNameWithoutExtension + "_receipt.txt");
+
+            await blockBlob.UploadTextAsync(receipt).ConfigureAwait(false);
+        }
+
+        public static string GetFileNameFromPath(string filePath)
+        {
+            if (filePath == null)
+            {
+                throw new ArgumentNullException(nameof(filePath));
+            }
+
+            var pathParts = new Uri(filePath).AbsolutePath.Split('/').ToList();
+            var cleanedPathParts = pathParts.SkipWhile(part => string.IsNullOrEmpty(part) || part.Equals("/", StringComparison.OrdinalIgnoreCase));
+
+            var fileName = string.Join('/', cleanedPathParts.Skip(1));
+            return fileName;
+        }
+
+        public static string GetFileNameWithoutExtension(string fileName)
+        {
+            if (fileName == null)
+            {
+                throw new ArgumentNullException(nameof(fileName));
+            }
+
+            var fileExtension = Path.GetExtension(fileName);
+
+            return fileName.Substring(0, fileName.Length - fileExtension.Length);
+        }
+
+        public static string GetContainerNameFromPath(string filePath)
+        {
+            if (filePath == null)
+            {
+                throw new ArgumentNullException(nameof(filePath));
+            }
+
+            var pathParts = new Uri(filePath).AbsolutePath.Split('/').ToList();
+            var cleanedPathParts = pathParts.SkipWhile(part => string.IsNullOrEmpty(part) || part.Equals("/", StringComparison.OrdinalIgnoreCase));
+
+            var containerName = cleanedPathParts.FirstOrDefault();
+            return containerName;
+        }
+
+        public static bool IsItAudio(string fileName)
+        {
+            string[] mediaExtensions =
+            {
+                ".WAV", ".MID", ".MIDI", ".WMA", ".MP3", ".OGG", ".RMA"
+            };
+
+            return mediaExtensions.Contains(Path.GetExtension(fileName).ToUpperInvariant());
```

Comment:
using file extension to get audio file type is not reliable and error prone.

Reply:
I agree. I removed the file extension check